### PR TITLE
Patch 2 - Living whitepaper update of ledger.md

### PR DESCRIPTION
**Pull request by qwahzi**
Pulling in original whitepaper language for ledger.md, along with additional clarifying details.
Added a Nano block example (output from block_create RPC command)
Added a ledger pruning blurb
Added a receive blocks blurb

### CODE REVIEW

File: docs/protocol-design/ledger.md [hunk 1]
```diff
@@ -6,16 +6,72 @@ Title: Nano Protocol Design - Ledger
...
+
 ### Blocks
 
+In traditional blockchain-based cryptocurrencies like Bitcoin, a block is a group of transactions. In Nano, a block is a single transaction, so the term “block” and “transaction” are often used interchangeably. "Transaction" specifically refers to the action, while block refers to the digital encoding of the transaction. Transactions are signed by the private-key belonging to the account on which the transaction is performed. 
+
+An example of a Nano block:
+```
+{
+  "hash": "FF0144381CFF0B2C079A115E7ADA7E96F43FD219446E7524C48D1CC9900C4F17",
+  "difficulty": "ffffffe1278b3dc6", // since V21.0
```
Comment (guilhermelawless) [inline docs/protocol-design/ledger.md:48]: These two are obtained from the details below, perhaps leave only the insides of "block" below

Comment (qwahzi) [reply docs/protocol-design/ledger.md]: Good point, fixed

**Comment by zhyatt** [comment]
@qwahzi Nice job getting the ledger page moving. I pushed in some adjustments for consideration, including merging in #400 for more consistent diagram usage (not sure if that will stick but TBD).

The more contentious piece of my updates is the terminology - you can see how I handled blocks and transaction definitions, and then added transfer in there to hopefully make things clearer. Curious how that language feels on your end. Perhaps it could be a discussion point you can have in Discord or the forum if you think it is worth some perspective.

**Comment by qwahzi** [comment]
> @qwahzi Nice job getting the ledger page moving. I pushed in some adjustments for consideration, including merging in #400 for more consistent diagram usage (not sure if that will stick but TBD).
> 
> The more contentious piece of my updates is the terminology - you can see how I handled blocks and transaction definitions, and then added transfer in there to hopefully make things clearer. Curious how that language feels on your end. Perhaps it could be a discussion point you can have in Discord or the forum if you think it is worth some perspective.

Thanks @zhyatt! I like the block vs transaction vs transfer breakdown, it makes sense to me and doesn't battle against any existing conventions in colloquial usage (that I know of).

**Comment by zhyatt** [comment]
@qwahzi Some further updates with new diagram for block lattice along with description expansion there, minor typo updates and a change of the callout box on top of the Blocks page. It is a duplicate of the definitions on the ledger page and might be useful as a standard way to do this elsewhere...thoughts?

**Comment by qwahzi** [comment]
> @qwahzi Some further updates with new diagram for block lattice along with description expansion there, minor typo updates and a change of the callout box on top of the Blocks page. It is a duplicate of the definitions on the ledger page and might be useful as a standard way to do this elsewhere...thoughts?

If it's included as a snippet, is it still needed under the Blocks header of the Ledger page? Probably fine either way though. Looks good!

**Comment by zhyatt** [comment]
> If it's included as a snippet, is it still needed under the Blocks header of the Ledger page? Probably fine either way though. Looks good!

There may be a better way to do this, so definitely open to ideas. The nav logically suggests the Ledger page should be reviewed first, and a generated PDF would have it first, so I think the terms make sense to have somewhere on that page to provide clarity. Currently they are in context on the Ledger page and then duplicated in the header of the following Blocks page since they apply there. Do you like the idea of a standard Key Terms section up top? Or perhaps on the Blocks page we could link back to the Ledger page for the origin definition to avoid duplicating it?

**Comment by qwahzi** [comment]
> > If it's included as a snippet, is it still needed under the Blocks header of the Ledger page? Probably fine either way though. Looks good!
> 
> There may be a better way to do this, so definitely open to ideas. The nav logically suggests the Ledger page should be reviewed first, and a generated PDF would have it first, so I think the terms make sense to have somewhere on that page to provide clarity. Currently they are in context on the Ledger page and then duplicated in the header of the following Blocks page since they apply there. Do you like the idea of a standard Key Terms section up top? Or perhaps on the Blocks page we could link back to the Ledger page for the origin definition to avoid duplicating it?

To be honest, I kind of like it as is. The ledger design is one of Nano's key features, but you have to understand accounts and blocks to understand the block-lattice. I don't think a separate key terms section is needed, since 1) the terms are already explained in the docs as they come up, and 2) the glossary still exists as a point of reference. 

Maybe we can merge it as is and then solicit feedback. Move fast, fail fast, improve fast :)

**Pull request merged**